classes/Cart.php

```diff
@@ -3140,7 +3140,7 @@ protected function getIdCarrierFromDeliveryOption($delivery_option)
     {
         $delivery_option_list = $this->getDeliveryOptionList();
         foreach ($delivery_option as $key => $value) {
-            if (isset($delivery_option_list[$key]) && isset($delivery_option_list[$key][$value])) {
+            if (isset($delivery_option_list[$key], $delivery_option_list[$key][$value])) {
```

Comment:
you can simplify this one with `isset($delivery_option_list[$key][$value])` only